### DISCUSSION TITLE
Merge non-overlapping maps

### DISCUSSION THREAD

**Discussion by albertarla** [stella-cv/stella_vslam#188] [Q&A] [answered]
Hello everybody,
I am currently trying to create a map of two zones joined by a corridor. I can map these two zones independently easily, but the corridor is impossible for me due to the light conditions and others. I have the relationship of the x, y, z coordinates of the two zones to the real latitude and longitude of these zones, with that I am able to relate them. Knowing this, I was thinking of some way to merge the maps of these two areas together in such a way that it gets lost when it goes to the corridor but finds itself again on the other map. I have a couple of ideas on how to deal with the problem but I wanted to know if someone has done it before or how you would do it. What I was thinking is to merge the two .msg files (I'm not sure how right now) or create another thread and when lost every thread will try to relocalize in its own map. Any other idea or what would be the best approach to do that?
Thanks in advance!

**Comment by ymd-stella** [comment, marked as answer]
It will also be necessary if you want to move between two levels in an elevator. I think there are several ways to do this. Rather than merging maps, I'm interested in dynamic loading, which can handle more extensive maps.

**Reply by albertarla** [reply]
Hi, thanks for the reply! How would you handle loading maps dynamically?I see some problems in that, for example in an elevator if there are more than one floor how would you handle wich map needs to be loaded? Also, which are the steps to follow to implement something like that? If there's anything I can do to implement it I would. I was making progress working with different threads of openvslam but I think that solution would need too many resources of my pc if there are more than two maps for example.

**Reply by ymd-stella** [reply]
> How would you handle loading maps dynamically?

I'm thinking of loading a second map into map_database. (We also need to think about deleting unnecessary maps if we don't have enough free memory.)

> Also, which are the steps to follow to implement something like that? If there's anything I can do to implement it I would.

Whether merging or loading dynamically, we need the relative poses (offsets) between multiple maps. The first question is how to calculate them. It may be enough to give a user-defined relative poses.

Next, we need to check if the system works fine when we load the second map into the map_database where we loaded the first map. If there are parts of the graph that are not connected, there may be a problem with some keyframe search. (See [graph_node](https://github.com/OpenVSLAM-Community/openvslam/blob/main/src/openvslam/data/graph_node.h))

**Reply by ymd-stella** [reply]
I will create an Issue.

**Reply by ymd-stella** [reply]
Let's consider the details in https://github.com/OpenVSLAM-Community/openvslam/issues/223.
